worlds/kdl3/__init__.py

```diff
@@ -129,6 +129,7 @@ def pre_fill(self) -> None:
             # randomize copy abilities
             valid_abilities = list(copy_ability_access_table.keys())
             enemies_to_set = list(self.copy_abilities.keys())
+            unplaced_abilities = set(key for key in copy_ability_access_table.keys() if key != "No Ability")
```

Comment:
this is just a guess so definitely worth checking, but depending on the size of this dictionary it might be a bit faster to just create the set then remove that specific ability rather than use a comprehension here. i like comprehensions though so it's already fine as is.